src/Microsoft.Extensions.Configuration/ConfigurationBuilder.cs

```diff
@@ -14,15 +15,11 @@ public class ConfigurationBuilder : IConfigurationBuilder
         private readonly IList<IConfigurationSource> _sources = new List<IConfigurationSource>();
 
         /// <summary>
-        /// Returns the sources used to obtain configuration values.
+        /// Gets or sets the source at the specified index.
         /// </summary>
-        public IEnumerable<IConfigurationSource> Sources
-        {
-            get
-            {
-                return _sources;
-            }
-        }
+        /// <param name="index">The index</param>
+        /// <returns>The source at the specified index.</returns>
+        public IConfigurationSource this[int index] { get => _sources[index]; set => _sources[index] = value; }
```

Comment:
Again, get rid of all these additions here and and just change the type of `Sources` to expose the underlying `List<IConfigurationSource>` that it already returns.